src/eth.rs

```diff
@@ -295,13 +295,9 @@ impl<'d> EthDriver<'d> {
     esp_idf_eth_spi_ethernet_ksz8851snl
 ))]
 impl<'d> EthDriver<'d> {
-    pub fn new_spi<P: spi::Spi + 'd>(
-        spi: impl Peripheral<P = P> + 'd,
+    pub fn new_spi(
+        spi_driver: &spi::SpiDriver<'d>,
```

Comment:
You are re-using the `SpiDriver` thing, but you are not re-using `SpiDeviceDriver`. The latter would allow you to get rid of `init_spi_device` and `get_spi_conf`. Since this is an obtrusive change, why not bite the bullet and pass-in an `SpiDeviceDriver` instance instead?

Reply:
Plan was to just change eth.rs API. Passing in `SpiDeviceDriver` instead of `SpiDriver` would mean that caller need to do all the configuration on his own. Wouldn't want to expose that complexity.

I do see how we could create `SpiDeviceDriver` internally to get rid of `init_spi_device` and `get_spi_conf`, but that can be done later as well.

Reply:
What additional configuration do you mean specifically? Please point me to it.

As for this being done later - I'm not feeling comfortable breaking the API twice just because. So we take either the current  route (`SpiDriver`) or what I suggested (`SpiDeviceDriver`). With good arguments for one or the other.